File: pset6 - Python/readability/readability.py
from cs50 import get_string


def main():

    text = get_string("Text:")

    print(coleman_liau(text))


# Determines the grade of the text passed as arguement

def coleman_liau(text):

    letters = 0
    words = 1
    sentences = 0
    text = text.lower()

    # Checks every character and adds it according to the specs
    for i in range(0, len(text)):
        if text[i] >= "a" and text[i] <= "z":
            letters += 1
        elif text[i] == " ":
            words += 1
        elif text[i] == "." or text[i] == "!" or text[i] == "?":
            sentences += 1

    # letters per 100 words
    L = (letters * 100) / words
    # sentences per 100 words
    S = (sentences * 100) / words

    grade = round(0.0588 * L - 0.296 * S - 15.8)

    # Return the grade
    if grade < 1:
        return "Before Grade 1"
    elif grade > 1 and grade < 16:
        return f"Grade {grade}"
    else:
        return "Grade 16+"


main()

File: pset2 - Arrays/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int colemanLiauIndex(string text);

int main(void)
{
    string text = get_string("Text: ");

    int grade = colemanLiauIndex(text);

    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade < 16)
    {
        printf("Grade %i\n", grade);
    }
    else
    {
        printf("Grade 16+\n");
    }
}

int colemanLiauIndex(string text)
{
    int letterSum = 0;
    float wordSum = 1;
    int sentenceSum = 0;

    // se repite el ciclo hasta que se recorre toda la longitud del string
    for (int i = 0; i < strlen(text); i++)
    {
        // inicialmente pasamos los caracteres a lowercase
        char character = tolower(text[i]);

        // si el caracter se encuentra entre a y z es una letra y se suma
        if ((character >= 'a') && (character <= 'z'))
        {
            letterSum++;
        }
        // si aparece un espacio asumimos que termino una palabra y comienza otra
        else if (character == ' ')
        {
            wordSum++;
        }
        // Si aparece alguno de estos tres caracteres asumimos que se termino una frase
        else if (character == '?' || character == '!' || character == '.')
        {
            sentenceSum++;

        }

    }

    //Calculo del indice en base a los valores de letterSum, wordSum y sentenceSum
    // index = 0.0588 * L - 0.296 * S - 15.8
    // L is the average number of letters per 100 words in the text, and S is the average number of sentences per 100 words in the text.
    float L = (letterSum * 100) / wordSum;
    float S = (sentenceSum * 100) / wordSum;

    int index = round(0.0588 * L - 0.296 * S - 15.8);

    return index;
}